pkg/provider/resource_project.go

```diff
@@ -80,73 +97,145 @@ func (p *ProjectResource) Schema() *schema.Resource {
+}
+
+type cloudProvider struct {
+	CloudProviderId   string `tfsdk:"cloud_provider_id"`
+	CloudProviderName string `tfsdk:"cloud_provider_name"`
+}
+
+type Project struct {
+	ID             *string         `tfsdk:"id"`
+	ProjectID      *string         `tfsdk:"project_id"`
+	ProjectName    *string         `tfsdk:"project_name"`
+	UserCount      *int            `tfsdk:"user_count"`
+	ClusterCount   *int            `tfsdk:"cluster_count"`
+	CloudProviders []cloudProvider `tfsdk:"cloud_providers"`
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (p projectResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var config Project
+	diags := req.Config.Get(ctx, &config)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 
-	projectId, err := client.Create(ctx, project_name)
+	projectId, err := p.client.Create(ctx, *config.ProjectName)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating project", "Could not create project, unexpected error: "+err.Error())
+		return
+	}
 
+	project, err := p.client.Read(ctx, projectId)
 	if err != nil {
-		return fromBigAnimalErr(err)
+		resp.Diagnostics.AddError("Error reading project", "Could not read project, unexpected error: "+err.Error())
+		return
 	}
 
-	d.SetId(projectId)
+	config.ID = &project.ProjectId
+	config.ProjectID = &project.ProjectId
+	config.ProjectName = &project.ProjectName
+	config.UserCount = &project.UserCount
+	config.ClusterCount = &project.ClusterCount
+	for _, provider := range project.CloudProviders {
+		config.CloudProviders = append(config.CloudProviders, cloudProvider{
+			CloudProviderId:   provider.CloudProviderId,
+			CloudProviderName: provider.CloudProviderName,
+		})
+	}
 
-	if err := p.read(ctx, d, meta); err != nil {
-		return diag.FromErr(err)
+	diags = resp.State.Set(ctx, &config)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
 	}
-	return diag.Diagnostics{}
 }
 
-func (p *ProjectResource) Read(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	if err := p.read(ctx, d, meta); err != nil {
-		return fromBigAnimalErr(err)
+// Read refreshes the Terraform state with the latest data.
+func (p projectResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state Project
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	project, err := p.client.Read(ctx, *state.ID)
+	if err != nil {
+		resp.Diagnostics.AddError("Error reading project", "Could not read project, unexpected error: "+err.Error())
+		return
+	}
+
+	state.ProjectID = &project.ProjectId
+	state.ProjectName = &project.ProjectName
+	state.UserCount = &project.UserCount
+	state.ClusterCount = &project.ClusterCount
+	state.CloudProviders = nil
+	for _, provider := range project.CloudProviders {
```

Comment:
This code block will add the cloud providers even if they are written to the state. We need to check if they already exist in the cloud providers section